src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,89 @@
+package duke;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+
+import java.time.LocalDate;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+/**
+ * The Storage class handles the read and write of data from and to the local disk.
+ */
+public class Storage {
+    /**
+     * Data file path.
+     */
+    private final String FILE_PATH;
+
+    /**
+     * Instantiates a Storage object with a file path.
+     * @param path File path to read from and write to.
+     */
+    Storage(String path) {
+        FILE_PATH = path;
+        try {
+            File f = new File(FILE_PATH);
+            if (!f.exists()) {
+                if (f.getParentFile().mkdirs() && f.createNewFile()) {
+                    System.out.println("\tStorage space for you tasks has been initialized successfully.");
+                }
+            }
+        } catch (IOException e) {
+            System.out.println("Oops! Something went wrong when loading your tasks from storage.");
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Reads saved data from the storage file and returns a list of {@link Task}s.
+     * @return A list of {@link Task}s.
+     * @throws IOException Exception when reading from storage file.
+     */
+    List<Task> processStorage() throws IOException {
```

Comment:
Should the method definition have an access modifier?

Reply:
Changed it to public!